api/v1alpha1/bgppeer_webhook.go

```diff
@@ -70,12 +75,15 @@ func (bgpPeer *BGPPeer) ValidateDelete() error {
 	return nil
 }
 
-func (bgpPeer *BGPPeer) validateBGPPeer(existingBGPPeersList *BGPPeerList) error {
+func (bgpPeer *BGPPeer) validateBGPPeer(existingBGPPeersList *BGPPeerList, bgpFrrMode bool) error {
 	var allErrs field.ErrorList
 
 	if err := bgpPeer.validateBGPPeersRouterID(existingBGPPeersList); err != nil {
 		allErrs = append(allErrs, err)
 	}
+	if err := bgpPeer.validateBGPPeersMyASN(existingBGPPeersList, bgpFrrMode); err != nil {
```

Comment:
can you move the if outside? Like
if bgpFrrMode {

}
I think it will be less confusing and keep the validation function more focused.